src/exit/free.c

```diff
@@ -38,6 +38,8 @@ void	free_data(t_data *data)
 		close(data->orig_fdin);
 	if (data->orig_fdout)
 		close(data->orig_fdout);
+	if (data->hd_fd)
+		close(data->hd_fd);
```

Comment:
Closing the heredoc file descriptor in case the heredoc process gets interrupted with SIGINT (Ctrl + C).